### PR TITLE
Add Examples

### PR DESCRIPTION
**Pull request by mwestphal**
- Fix an issue with PUBLIC link in public
- Fix a potential segfault with an invalid file
- Add examples
- Use examples instead of test_sdk to validate install

### CODE REVIEW

**Comment by mwestphal** [comment]
data issue resolved, needs macos fix and rendering fix to go forward

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -193,25 +194,36 @@ jobs:
...
+      if: matrix.os == 'windows-latest'
+      uses: ./source/.github/actions/mesa-install
+      with:
+        path: ${{github.workspace}}\build_examples\render-image\Release\
+
+    - name: Install Mesa Windows Examples2
+      if: matrix.os == 'windows-latest'
+      uses: ./source/.github/actions/mesa-install
+      with:
+        path: ${{github.workspace}}\build_examples\render-interact\Release\
```
Comment (mwestphal) [inline .github/workflows/ci.yml:213]: I could not find a better way to fix this

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -129,7 +130,7 @@ jobs:
         -DF3D_BINDINGS_JAVA=ON
         -DF3D_INSTALL_SDK=ON
         -DF3D_INSTALL_SDK_PLUGIN=ON
-        -DCMAKE_BUILD_TYPE=Release
+        -DCMAKE_BUILD_TYPE=Debug
```
Comment (Meakk) [inline .github/workflows/ci.yml:133]: why this change?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -193,25 +194,38 @@ jobs:
...
+        path: ${{github.workspace}}\build_examples\render-interact\Release\
+
+    # older VTK version create different renderings so the are not tested
+    - name: Test libf3d examples
+      if: matrix.vtk_version != 'v9.0.0' && matrix.vtk_version != 'v9.1.0'
+      working-directory: ${{github.workspace}}
+      run: ctest --test-dir build_examples -C Release -VV
+
+    # Windows do not support shared VTK in CI yet
+    - name: Validate plugin exaples
```
Comment (Meakk) [inline .github/workflows/ci.yml:222]:
```suggestion
    - name: Validate plugin examples
```

File: cmake/f3dPlugin.cmake [hunk 4]
```diff
@@ -243,7 +243,7 @@ macro(f3d_plugin_build)
 
   list(TRANSFORM F3D_PLUGIN_VTK_MODULES PREPEND "VTK::")
 
-  target_link_libraries(f3d-plugin-${F3D_PLUGIN_NAME} PUBLIC
+  target_link_libraries(f3d-plugin-${F3D_PLUGIN_NAME} PRIVATE
```
Comment (Meakk) [inline cmake/f3dPlugin.cmake:246]: Why is it needed?

File: examples/README.md [hunk 5]
```diff
@@ -1,3 +1,23 @@
...
+
+Examples plugins for the libf3d, which requires the `F3D_INSTALL_PLUGIN_SDK` cmake option when building F3D.
+
+## Building
+
+Each example is intended to be built as a separate CMake project, however top level `CMakeLists.txt` are provided if needed.
+
+```
+cmake -Df3d_DIR=/path/to/f3d/install/lib/cmake/f3d /path/to/example/dir
+cmake --build
```
Comment (Meakk) [inline examples/README.md:20]:
```suggestion
cmake --build .
```

File: examples/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,12 @@
+project(examples)
+
+cmake_minimum_required(VERSION 3.2)
```
Comment (Meakk) [inline examples/CMakeLists.txt:3]: Are we sure about this version?

File: examples/plugins/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,11 @@
+project(libf3d-examples)
+
+cmake_minimum_required(VERSION 3.2)
+
+find_package(f3d REQUIRED)
```
Comment (Meakk) [inline examples/plugins/CMakeLists.txt:5]: Check for `f3d_SDK_PLUGIN`

**Review by Meakk — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (mwestphal) [reply .github/workflows/ci.yml:133]: should have been removed, nice find

File: cmake/f3dPlugin.cmake (hunk 4 above)
Comment (mwestphal) [reply cmake/f3dPlugin.cmake:246]: actually not needed, but you mentionned that it was a mistake for it to be public

File: examples/CMakeLists.txt (hunk 6 above)
Comment (mwestphal) [reply examples/CMakeLists.txt:3]: Not at all, do you want to find out the min version ?

File: cmake/f3dPlugin.cmake (hunk 4 above)
Comment (Meakk) [reply cmake/f3dPlugin.cmake:246]: It does not make any difference here I think since there is no linkage to plugin libraries.

File: examples/CMakeLists.txt (hunk 6 above)
Comment (Meakk) [reply examples/CMakeLists.txt:3]: Not necessary in this PR, but it would be great to find out what version we are really using.

File: cmake/f3dPlugin.cmake (hunk 4 above)
Comment (mwestphal) [reply cmake/f3dPlugin.cmake:246]: since there is no reason for it to be pulic, lets keep it private ?

File: examples/CMakeLists.txt (hunk 6 above)
Comment (mwestphal) [reply examples/CMakeLists.txt:3]: done in this PR, libf3d is cmake 3.1 (or earlier, did not check cmake 2), plugins and F3D are 3.13. (` Unknown CMake command "target_link_options".`
